src/coreComponents/mesh/ParticleManager.cpp

```diff
@@ -0,0 +1,347 @@
+
+      unpackedSize += subRegion.unpack( buffer, particleList, 0, false, events );
+    } );
+  }
+
+  waitAllDeviceEvents( events );
+  return unpackedSize;
+}
+
+int ParticleManager::PackGlobalMapsSize( ParticleViewAccessor< arrayView1d< localIndex > > const & packList ) const
+{
+  buffer_unit_type * junk = nullptr;
+  return PackGlobalMapsPrivate< false >( junk, packList );
+}
+
+int ParticleManager::PackGlobalMaps( buffer_unit_type * & buffer,
+                                          ParticleViewAccessor< arrayView1d< localIndex > > const & packList ) const
+{
+  return PackGlobalMapsPrivate< true >( buffer, packList );
+}
+template< bool DOPACK >
+int
+ParticleManager::PackGlobalMapsPrivate( buffer_unit_type * & buffer,
+                                             ParticleViewAccessor< arrayView1d< localIndex > > const & packList ) const
+{
+  int packedSize = 0;
+
+  packedSize += bufferOps::Pack< DOPACK >( buffer, numRegions() );
+
+  for( typename dataRepository::indexType kReg=0; kReg<numRegions(); ++kReg )
+  {
+    ParticleRegionBase const & particleRegion = getRegion( kReg );
+    packedSize += bufferOps::Pack< DOPACK >( buffer, particleRegion.getName() );
+
+    packedSize += bufferOps::Pack< DOPACK >( buffer, particleRegion.numSubRegions() );
+    particleRegion.forParticleSubRegionsIndex< ParticleSubRegionBase >(
+      [&]( localIndex const esr, ParticleSubRegionBase const & subRegion )
+    {
+      packedSize += bufferOps::Pack< DOPACK >( buffer, subRegion.getName() );
+
+      arrayView1d< localIndex const > const particleList = packList[kReg][esr];
+      if( DOPACK )
+      {
+        packedSize += subRegion.packGlobalMaps( buffer, particleList, 0 );
+      }
+      else
+      {
+        packedSize += subRegion.packGlobalMapsSize( particleList, 0 );
+      }
+    } );
+  }
+
+  return packedSize;
+}
+
+
+int
+ParticleManager::UnpackGlobalMaps( buffer_unit_type const * & buffer,
+                                        ParticleViewAccessor< ReferenceWrapper< localIndex_array > > & packList )
+{
+  int unpackedSize = 0;
+
+  localIndex numRegionsRead;
+  unpackedSize += bufferOps::Unpack( buffer, numRegionsRead );
+
+  packList.resize( numRegionsRead );
+  for( localIndex kReg=0; kReg<numRegionsRead; ++kReg )
+  {
+    string regionName;
+    unpackedSize += bufferOps::Unpack( buffer, regionName );
+
+    ParticleRegionBase & particleRegion = getRegion( regionName );
+
+    localIndex numSubRegionsRead;
+    unpackedSize += bufferOps::Unpack( buffer, numSubRegionsRead );
+    packList[kReg].resize( numSubRegionsRead );
+    particleRegion.forParticleSubRegionsIndex< ParticleSubRegionBase >(
+      [&]( localIndex const esr, ParticleSubRegionBase & subRegion )
+    {
+      string subRegionName;
+      unpackedSize += bufferOps::Unpack( buffer, subRegionName );
+
+      /// THIS IS WRONG
+      localIndex_array & particleList = packList[kReg][esr].get();
```

Comment:
what's wrong? I guess it should be fixed before merging this into develop. Or, at least, I would add a detailed comment of what is not as you would like it to be.

Reply:
This just got copied over from https://github.com/GEOS-DEV/GEOS/blob/414008af660eacbc7836fd39faf2c9b116bec237/src/coreComponents/mesh/ElementRegionManager.cpp#L443, I figured it was there for a reason

Reply:
That comment was added when the DEM solver and manager in old geos was being used as a basis for the particle data for the MPM code.  The MPM code didn't use global maps, but i recall during debugging when we were comparing packed and unpacked data that there was some inconsistency related to the global maps.  The issue may have related to the presence of ghost particles in the data structure.  

I've added a more descriptive comment in the code.  we may wish to create a unit test at some point to investigate.